src/Fixer/PhpTag/ShortEchoTagFixer.php

```diff
@@ -0,0 +1,242 @@
+                ->getOption(),
+            (new FixerOptionBuilder(self::OPTION_LONG_FUNCTION, 'The function to be used to expand the short echo tags'))
+                ->setAllowedValues($this->supportedLongFunctionOptions)
+                ->setDefault(self::LONG_FUNCTION_ECHO)
+                ->getOption(),
+            (new FixerOptionBuilder(self::OPTION_SHORT_ALWAYS, 'Always render short-echo tags even in case of complex code'))
+                ->setAllowedTypes(['bool'])
+                ->setDefault(false)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(SplFileInfo $file, Tokens $tokens)
+    {
+        if (self::FORMAT_SHORT === $this->configuration[self::OPTION_FORMAT]) {
+            $this->longToShort($file, $tokens);
+        } else {
+            $this->shortToLong($file, $tokens);
+        }
+    }
+
+    private function longToShort(SplFileInfo $file, Tokens $tokens)
+    {
+        $offset = 0;
+        for (;;) {
+            $foundEcho = $tokens->findSequence([[\T_OPEN_TAG], [\T_ECHO]], $offset);
+            $foundPrint = $tokens->findSequence([[\T_OPEN_TAG], [\T_PRINT]], $offset);
+            if (null === $foundEcho) {
+                $found = $foundPrint;
+                if (null === $found) {
+                    break;
+                }
+            } elseif (null === $foundPrint || key($foundEcho) < key($foundPrint)) {
+                $found = $foundEcho;
+            } else {
+                $found = $foundPrint;
+            }
+            $startRange = key($found);
+            $offset = $startRange + 1;
+            if (!$this->configuration[self::OPTION_SHORT_ALWAYS] && $this->isComplexCode($tokens, $offset)) {
+                continue;
+            }
+            next($found);
+            $endRange = key($found);
+            $newTokens = $this->buildLongToShortTokens($tokens, $startRange, $endRange);
+            $tokens->overrideRange($startRange, $endRange, $newTokens);
+        }
+    }
+
+    private function shortToLong(SplFileInfo $file, Tokens $tokens)
+    {
+        if (self::LONG_FUNCTION_PRINT === $this->configuration[self::OPTION_LONG_FUNCTION]) {
+            $echoToken = [\T_PRINT, 'print'];
+        } else {
+            $echoToken = [\T_ECHO, 'echo'];
+        }
+        $offset = $tokens->count() - 1;
+        for (;;) {
+            $found = $tokens->getPrevTokenOfKind($offset, [[\T_OPEN_TAG_WITH_ECHO]]);
+            if (null === $found) {
+                break;
+            }
+            $offset = $found - 1;
+            $replace = [new Token([\T_OPEN_TAG, '<?php ']), new Token($echoToken)];
+            if (!$tokens[$found + 1]->isWhitespace()) {
+                $replace[] = new Token([\T_WHITESPACE, ' ']);
+            }
+            $tokens->overrideRange($found, $found, $replace);
+        }
+    }
+
+    private function isComplexCode(Tokens $tokens, $index)
```

Comment:
I'm not sure id I understand it - can you add few more tests for this option?

Reply:
There are already tests for it. BTW I added a PHPDoc comment to this method: dadc11e152f13bb15854f57ce4dc5eccf8c99f13